modules/lib/default.nix

```
@@ -0,0 +1,86 @@
+lib:
+with builtins;
+with lib; rec {
+  # string -> string
+  # this capitalizes the first letter in a string,
+  # which is sometimes needed in order to format
+  # the names of themes correctly
+  mkUpper = word:
+    (toUpper (substring 0 1 word)) + (substring 1 (stringLength word) word);
```

```suggestion
  mkUpper = str:
    (toUpper (substring 0 1 str)) + (substring 1 (stringLength str) str);
```
`word` doesn't make much sense now that this is a library function.